common/persistence/sql/sqlplugin/postgres/task.go

```diff
@@ -48,7 +48,8 @@ task_type = :task_type
 
 	listTaskListQry = `SELECT domain_id, range_id, name, task_type, data, data_encoding ` +
 		`FROM task_lists ` +
-		`WHERE shard_id = $1 AND domain_id > $2 AND name > $3 AND task_type > $4 ORDER BY domain_id,name,task_type LIMIT $5`
+		`WHERE shard_id = $1 AND ((domain_id = $2 AND name = $3 AND task_type > $4) OR (domain_id=$2 AND name > $3) OR (domain_id > $2)) ` +
```

Comment:
any reason this is only changed for Postgres, not MySQL?

Reply:
Can you add more context/thoughts about this change. Even though the existing one looks broken to me. But I still have a hard time to understand why this new one will work.

Reply:
Just an oversight; we actively use the postgres path and not the mysql one.

Reply:
I see. thanks.